### PR TITLE
Fix compiler warnings: 'foo' has virtual functions but non-virtual destructor

### PR DESCRIPTION
**Pull request by ksooo**
... just what the title says.

### CODE REVIEW

**Comment by AlwinEsch** [comment]
Thats good to have, during addon tests with my new interface I seen also that the parent-parent destructor was not called without them on parent of child.

Mean like:

| CMyAddon -> | CInstanceScreensaver -> | IAddonInstance |
|:--------------------|:------------------------------------|:---------------------|
| no matter | if destructor missing | present desctructor not called

File: xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.h [hunk 1]
```diff
@@ -50,6 +50,8 @@ class CRenderManager;
 class IRenderMsg
 {
   friend CRenderManager;
+public:
+  virtual ~IRenderMsg() = default;
```
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.h]: declaring a dtor on an interface looks weird. there is nothing to destruct.

Comment (ksooo) [inline xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.h]: I guess you are right for interfaces (== classes with only pure virtual member functions). Will remove the respective changes from this PR.

Comment (ksooo) [inline xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.h]: Well, maybe it is actually a good idea (if not needed) to add a default virtual ctor, even to a c++ interface, as the compiler may generate a non-virtual dtor for that class:

<pre>
class IBatch
{
    public:
       virtual void f() = 0;
};

class A : public IBatch
{
    public:
       void f() {}
       ~A() {}
};

IBatch * a = new A();
a->f();    // calls A::f()
delete a;  // calls IBatch::~IBatch() not A::~A()
</pre> 

Source: https://stackoverflow.com/questions/2691611/destructors-for-c-interface-like-classes

Comment (ksooo) [inline xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.h]: @paxxi ^^^?

Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.h]: good catch. i think we should check all of our interface classes for this.

Comment (ksooo) [inline xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.h]: I did, on macOS, using the respective clang warning. So, only non-Mac classes still need to be checked.

Comment (pkerling) [inline xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.h]: Do note that this is only relevant if you are (potentially) going to destroy the class via a pointer to the interface.

Comment (ksooo) [inline xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.h]: > Do note that this is only relevant if you are (potentially) going to destroy the class via a pointer to the interface.

 ofc, but as you never know before and imo nobody checks the presence of a virtual dtor when first heap allocating a respective object and the price is just one more vtbl entry I tend to add this everywhere and be on the safe side.

**Comment by ksooo** [comment]
jenkins build this please

**Comment by ksooo** [comment]
jenkins error is unrelated

File: xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.h (hunk 1 above)
Comment (Paxxi) [inline xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.h]: sorry about the late response, good catch ksoo, fully approve

**Pull request merged**